src/thunderbots/software/network_input/filter/robot_filter.cpp

```diff
@@ -1,19 +1,94 @@
 #include "network_input/filter/robot_filter.h"
 
-RobotFilter::RobotFilter(unsigned int id) : robot_id(id) {}
+RobotFilter::RobotFilter(unsigned int id, Duration expiry_buffer_duration)
+    : robot_id(id), expiry_buffer_duration(expiry_buffer_duration)
+{
+}
 
-FilteredRobotData RobotFilter::getFilteredData(
+std::optional<Robot> RobotFilter::getFilteredData(
+    std::optional<Robot> current_robot_state,
     const std::vector<SSLRobotDetection> &new_robot_data)
 {
+    // if current robot state is not initialized, initialized it
+    if (!current_robot_state.has_value())
+    {
+        current_robot_state.emplace(Robot(this->getRobotId(),Point(),Vector(),Angle(),AngularVelocity(),Timestamp()));
```

Comment:
I believe it is usually good practice to initialise with values. Same goes for the code bellow.

Reply:
Changed. Initialised with values.